src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/TestUtil.java

```diff
@@ -237,4 +255,23 @@ public static int adjustFlushCountForOutputStreamClose(int flushCount) {
         return result;
     }
 
+    /**
+     * Verifies that a task with an expected result can be executed with
+     * limited stack size and time duration.
+     *
+     * @param <V> return type of task
+     * @param callable the task to execute
+     * @return true if task completes
+     * @throws InterruptedException if thread execution is interrupted
+     * @throws ExecutionException if task fails
+     */
+    public static <V> boolean isExecutableWithLimitedResources(Callable<V> callable)
+            throws InterruptedException, ExecutionException {
+        final FutureTask<V> futureTask = new FutureTask<>(callable);
+        final Thread thread = new Thread(null, futureTask,
+                "LimitedStackSizeThread", MINIMAL_STACK_SIZE);
+        thread.start();
+        thread.join(EXECUTION_TIMEOUT);
+        return futureTask.get() != null;
```

Comment:
```
thread.join(EXECUTION_TIMEOUT);
        return futureTask.get() != null;
```
looks better if changed to
`futureTask.get(10, TimeUnit.SECONDS) != null;`

Reply:
Great, thanks for this.